lib/plugins/aws/provider/awsProvider.js

```diff
@@ -470,7 +470,13 @@ class AwsProvider {
   }
 
   getDeploymentPrefix() {
-    return this.serverless.service.provider.deploymentPrefix || 'serverless';
+    const provider = this.serverless.service.provider;
```

Comment:
I think right way, is to agree it's a _string_ property (with no special support for booleans), and simply fallback to `serverless` on any no-value (so `null` or `undefined`)

It can be achieved simply as:

```javascript
const customPrefix = this.serverless.service.provider.deploymentPrefix;
return customPrefix == null ? 'serverless' : customPrefix
```

Reply:
I guess that the main idea is to allow `deploymentPrefix: false` as a more clear alternative to `deploymentPrefix: ''`. But, if you prefer to not do that, I can do that.

Reply:
`deploymentPrefix` is a _string_ property, so `false` should translate to `'false'` prefix (which technically should also be possible to set if someone wants it).

So, yes, I believe `customPrefix == null ? 'serverless' : customPrefix` is the right way to go.